### ISSUE TITLE
"free(): invalid pointer" and abnormal termination on shutdown

### ISSUE THREAD

**Issue by stefan-sherwood** [openDsh/dash#69]
class `Config` is constructed with `QApplication` as its parent (in `src/app/config.cpp`).

```
Config::Config()
    : QObject(qApp),
```
but it is instantiated on the stack

```
Config *Config::get_instance()
{
    static Config config;
    return &config;
}
```

On shutdown, `QApplication` calls `free()` on all of its children, including the instance of `Config`. This causes an abnormal program termination on every shutdown.

**Comment by rsjudka** [comment]
yeah `Config` was actually supposed to go away on my last major rework, but there were some open PRs still using it so I decided to leave it for now

Once I get to cleaning that up, it'll be gone 😄

**Comment by rsjudka** [comment]
there's also a bug in the OA lib when you shut down, so I've also been a bit more laxed on a clean dash shutdown for now :p
